File: leetcode/python/0035_search_insert_position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        l, r, middle = 0, len(nums) - 1, 0

        while (l <= r):
            middle = l + (r - l) / 2

            if target < nums[middle]:
                r = middle - 1
            elif target > nums[middle]:
                l = middle + 1
            else:
                return middle
        
        return l

File: leetcode/java/0035_search_insert_position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int l = 0, r = nums.length - 1, middle = 0;

        while (l <= r) {
            middle = l + ((r - l) / 2);

            if (target > nums[middle]) {
                l = middle + 1;
            } else if (target < nums[middle]) {
                r = middle - 1;
            } else {
                return middle;
            }
        }

        return l;
    }
}